docs/strings.rst

```diff
@@ -58,13 +63,38 @@ In either case, null bytes (``'\x00'``) in the data will cause an error.
    Fixed-length strings in HDF5 hold a set number of bytes.
    It may take multiple bytes to store one character.
 
+.. _npystrings:
+
+NumPy variable-width strings
```

Comment:
This section also needs an update - datasets created with 'T' dtype are now regular variable-length string datasets, and reopening them doesn't doesn't matter.